src/nuget.ts

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as https from 'https';
+import { Logger } from './logger';
+import * as path from 'path';
+import { PlatformInformation } from './platform';
+import * as proc from 'process';
+import { getProxyAgent } from './proxy';
+import { parse as parseUrl } from 'url';
+import * as xmlParser from 'xml-parser';
+
+export interface PackageSource {
+    source: string;
+    indexUrl: string;
+}
+
+export interface NuGetService {
+    autoCompleteService: string[];
+    packageRegistrationService: string;
+}
+
+export class NuGetClient{
+
+    constructor(private platformInfo: PlatformInformation) { }
+
+    public PackageSources: PackageSource[] = [];
+    public NugetServices: { [id: string]: NuGetService } = {};
+
+    public UpdatePackageSourcesFromConfig(logger: Logger): Promise<boolean> {
+        let packageSourcesUpdated = false;
+        return new Promise<boolean>((resolve, reject) => {
+            let nuGetConfig: string;
+
+            if (this.platformInfo.isWindows) {
+                nuGetConfig = path.join(proc.env.AppData, 'NuGet', 'NuGet.config');
+            } else {
+                // found this on StackOverflow (http://stackoverflow.com/questions/30076371/mac-osx-mono-nuget-how-to-set-up-a-default-appdata-nuget-nuget-config)
+                // and this http://gunnarpeipman.com/2014/10/running-asp-net-5-on-linux/
+                // TODO: verify the file actually resides there
+                nuGetConfig = path.join('~', '.config', 'NuGet', 'NuGet.config');
+            }
+
+            logger.appendLine('using active feeds from ' + nuGetConfig);
+
+            fs.readFile(nuGetConfig, 'utf-8', (err, data) => {
+                if (err) {
+                    reject('Reading ' + nuGetConfig + ' failed with error ' + err);
```

Comment:
Note that this will just carry below trying to access `data`. I suspect you want to return here.